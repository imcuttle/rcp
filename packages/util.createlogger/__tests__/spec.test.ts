/**
 * @file spec
 * @author imcuttle
 * @description
 */
process.env.DEBUG = 'scope'
import createLogger from '../'

describe('logger', function() {
  it('should format', () => {
    const c = createLogger()
    expect(c.format({ message: 'msss', type: 'aa', argv: [] })).toMatchSnapshot()
    expect(c.format({ message: 'msss:%s %o', type: 'aa', argv: ['abc', { ac: 'sss' }] })).toMatchSnapshot()
  })

  it('should format with scope', () => {
    const c = createLogger('scope')
    expect(c.format({ message: 'msss', type: 'aa', argv: [] })).toMatchSnapshot()
    expect(c.format({ message: 'msss:%s %o', type: 'aa', argv: ['abc', { ac: 'sss' }] })).toMatchSnapshot()
  })

  it('should debug with scope', () => {
    const c = createLogger('scope')
    c.log('saa%s + %s', 'as', 'ssw')
    c.warn('saa%s + %s', 'as', 'ssw')
    c.error('saa%s + %s', 'as', 'ssw')
    c.debug('saa%s + %s', 'as', 'ssw')
  })
})
