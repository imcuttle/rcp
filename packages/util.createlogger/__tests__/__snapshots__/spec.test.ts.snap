// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger should format 1`] = `"aa: msss"`;

exports[`logger should format 2`] = `"aa: msss:abc  { ac: 'sss' }"`;

exports[`logger should format in browserCreateLogger 1`] = `"aa: msss"`;

exports[`logger should format in browserCreateLogger 2`] = `"aa: msss:abc  { ac: 'sss' }"`;

exports[`logger should format with scope 1`] = `"aa(scope): msss"`;

exports[`logger should format with scope 2`] = `"aa(scope): msss:abc  { ac: 'sss' }"`;

exports[`logger should format with scope in browserCreateLogger 1`] = `"aa(scope): msss"`;

exports[`logger should format with scope in browserCreateLogger 2`] = `"aa(scope): msss:abc  { ac: 'sss' }"`;
